File: app/src/main/java/io/treehouses/remote/Network/ParseDbService.java
package io.treehouses.remote.Network;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Build;

import com.parse.ParseObject;

import java.util.HashMap;

import io.treehouses.remote.utils.Utils;
import io.treehouses.remote.utils.VersionUtils;

public class ParseDbService {
    public ParseDbService() {
    }

    public static void sendLog(Context context, String rpiName, HashMap<String, String> map, SharedPreferences preferences) {
        ParseObject testObject = new ParseObject("userlog");
        testObject.put("title", rpiName + "");
        testObject.put("description", "Connected to bluetooth");
        testObject.put("type", "BT Connection");
        testObject.put("versionCode", VersionUtils.getVersionCode(context));
        testObject.put("versionName", VersionUtils.getVersionName(context));
        testObject.put("deviceName", Build.DEVICE);
        testObject.put("deviceManufacturer", Build.MANUFACTURER);
        testObject.put("deviceModel", Build.MODEL);
        testObject.put("deviceSerialNumber", Utils.getAndroidId(context));
        testObject.put("macAddress", Utils.getMacAddr());
        testObject.put("androidVersion", Build.VERSION.SDK_INT + "");
        testObject.put("gps_latitude", preferences.getString("last_lat", ""));
        testObject.put("gps_longitude", preferences.getString("last_lng", ""));
        testObject.put("imageVersion", map.get("imageVersion") + "");
        testObject.put("treehousesVersion", map.get("treehousesVersion") + "");
        testObject.put("bluetoothMacAddress", map.get("bluetoothMacAddress") + "");
        testObject.saveInBackground();
    }

}

File: app/src/main/kotlin/io/treehouses/remote/network/ParseDbService.kt
package io.treehouses.remote.network

import android.content.Context
import android.content.SharedPreferences
import android.os.Build
import com.parse.ParseObject
import io.treehouses.remote.utils.Utils.macAddr
import io.treehouses.remote.utils.VersionUtils.getVersionCode
import io.treehouses.remote.utils.VersionUtils.getVersionName
import java.util.*

object ParseDbService {
    fun sendLog(context: Context?, rpiName: String, map: HashMap<String, String?>, preferences: SharedPreferences) {
        val testObject = ParseObject("userlog")
        testObject.put("title", rpiName + "")
        testObject.put("description", "Connected to bluetooth")
        testObject.put("type", "BT Connection")
        testObject.put("versionCode", getVersionCode(context))
        testObject.put("versionName", getVersionName(context))
        testObject.put("deviceName", Build.DEVICE)
        testObject.put("deviceManufacturer", Build.MANUFACTURER)
        testObject.put("deviceModel", Build.MODEL)
        testObject.put("macAddress", macAddr)
        testObject.put("androidVersion", Build.VERSION.SDK_INT.toString() + "")
        testObject.put("gps_latitude", preferences.getString("last_lat", "")!!)
        testObject.put("gps_longitude", preferences.getString("last_lng", "")!!)
        testObject.put("imageVersion", map["imageVersion"].toString() + "")
        testObject.put("treehousesVersion", map["treehousesVersion"].toString() + "")
        testObject.put("bluetoothMacAddress", map["bluetoothMacAddress"].toString() + "")
        testObject.saveInBackground()
    }
    fun sendFeedback(map: HashMap<String, String>) {
        val obj = ParseObject("feedback")
        obj.put("name", map["name"].toString() + "")
        obj.put("email", map["email"].toString() + "")
        obj.put("phoneNumber", map["phoneNumber"].toString() + "")
        obj.put("feedbackType", map["feedbackType"].toString() + "")
        obj.put("message", map["message"].toString() + "")
        obj.saveEventually()
    }
}